app/src/main/java/io/github/droidkaigi/confsched2017/view/fragment/SessionsFragment.java

```diff
@@ -156,11 +158,57 @@ private void initView() {
         adapter = new SessionsAdapter(getContext());
         binding.recyclerView.setAdapter(adapter);
 
+        GestureDetector gestureDetector = new GestureDetector(getContext(), new GestureDetector.OnGestureListener() {
+            private boolean ignoreMotionEventTillDown = false;
+
+            @Override
+            public boolean onDown(MotionEvent motionEvent) {
+                ignoreMotionEventTillDown = true;
+                return false;
+            }
+
+            @Override
+            public void onShowPress(MotionEvent motionEvent) {
+            }
+
+            @Override
+            public boolean onSingleTapUp(MotionEvent motionEvent) {
+                return false;
+            }
+
+            @Override
+            public boolean onScroll(MotionEvent event1, MotionEvent event2, float v, float v1) {
+
+                // Send cancel event for item clicked when horizontal scrolling.
+                if (ignoreMotionEventTillDown) {
+                    final long now = SystemClock.uptimeMillis();
+                    MotionEvent cancelEvent = MotionEvent.obtain(now, now,
+                            MotionEvent.ACTION_CANCEL, 0.0f, 0.0f, 0);
+                    binding.recyclerView.forceToDispatchTouchEvent(cancelEvent);
+                    ignoreMotionEventTillDown = false;
+                }
+
+                return false;
+            }
+
+            @Override
+            public void onLongPress(MotionEvent motionEvent) {
+            }
+
+            @Override
+            public boolean onFling(MotionEvent motionEvent, MotionEvent motionEvent1, float v, float v1) {
+                return false;
+            }
+        });
+
         binding.root.setOnTouchListener((v, event) -> {
+            boolean isHandle = gestureDetector.onTouchEvent(event);
+
             MotionEvent e = MotionEvent.obtain(event);
             e.setLocation(e.getX() + binding.root.getScrollX(), e.getY() - binding.headerRow.getHeight());
             binding.recyclerView.forceToDispatchTouchEvent(e);
-            return false;
+
+            return isHandle;
```

Comment:
Why did you use the return value of new listener?
HorizontalScrolView's touch event `must` not be consumed.